example/zfs-luks-raidz.nix

```diff
@@ -0,0 +1,148 @@
+# meant for NAS setups with raidzX approach.
+{ ... }:
+let
+  raid_disks = {
+    "TOSHIBA_1360A003FVGG" = "/dev/disk/by-id/ata-TOSHIBA_MG08ACA16TE_1360A003FVGG";
+    "TOSHIBA_1360A00BFVGG" = "/dev/disk/by-id/ata-TOSHIBA_MG08ACA16TE_1360A00BFVGG";
+    "TOSHIBA_1360A00VFVGG" = "/dev/disk/by-id/ata-TOSHIBA_MG08ACA16TE_1360A00VFVGG";
+    "TOSHIBA_41R0A0EBF57H" = "/dev/disk/by-id/ata-TOSHIBA_MG08ACA16TE_41R0A0EBF57H";
+    "TOSHIBA_5120A03WF57H" = "/dev/disk/by-id/ata-TOSHIBA_MG08ACA16TE_5120A03WF57H";
+  };
```

Comment:
We usually make those over-writable for the test. The NixOS test framework allows to create additional disks.

Reply:
Like this : 
```
{
  root_disk ? "/dev/vdb",
  raid_disks ? {
    # better use /dev/disk/by-id/
    "vendor_id02" = "/dev/vdc";
    "vendor_id03" = "/dev/vdd";
    "vendor_id04" = "/dev/vde";
    "vendor_id05" = "/dev/vdf";
    "vendor_id06" = "/dev/vdg";
  }, ... }:
```

Reply:
Yes. We can also make it just one list and assume that disks[0] is the root disk? No strong opinion here.

Reply:
Hmm doing this kinda make the example even more useless, because naming your devices based on your vendor is part of a zfs-raid strategy to know which hard-drive needs to be changed if it failes. Of course you can always find your drive by investigating `/dev/disk/by-id` and `/dev/mapper`, but the idea here was to make this obsolete right from the start.

Reply:
you can have the default value still be /dev/disk/by-id. the test runner passes the disks as disks to the example, so they would get override, root_disk has to be disks[0] though

Reply:
Yeah for the root pool this might work, but not for the raid pool. This way I avoid to format the same partition twice.
For the raid pool I need a map. Hmm maybe I could use `builtins.basenameOf` to get rid of the map. Will think about it, ... at the moment it seems the test will never run anyway.